getting_started/step_by_step/debugger.rst

```diff
@@ -0,0 +1,119 @@
+.. Intention:
+    Introduce break points in debugger. It will help beginners to trace their codes.
+    
+.. Todo:
+    All the contents about C# is not included. Author is not familiar with C#.
+
+.. _doc_trace_code_with_debugger_:
+
+Using debugger
+==============
+
+In this lesson, we will look at the break points in Godot built-in debugger.
+We will write a bottom line checker to check if the position of Sprite2D node is 
+below bottom line. Bottom line checker will send Sprite2D node back to desired 
+position if it's under the bottom line.
+Also, we will set break points to see if the checker is being called after the 
+Sprite2D node crossed the bottom line.
+
+.. note:: More detailed introduction to debugger can be found in page 
+          :ref:`doc_debugger_panel`
+
+Setup scene
+-----------
+
+In :ref:`doc_scripting_player_input`, you should be able to control Sprite2D node 
+via keyboard arrows. Now, we need to make sure that Sprite2D node will not cross 
+the bottom line. Once the bottom line is crossed, we need to sent it back.
+
+First, we need to define a custom signal. Once the bottom line checker found that
+Sprite2D node crossed the bottom line, it will emit the signal. 
```

Comment:
```suggestion
First we need to define a custom signal. Once the bottom line checker has found that
the ``Sprite2D`` node has crossed the bottom line, it will emit the signal.
```
Or
```suggestion
First we need to define a custom signal. Once the bottom line checker finds that
the ``Sprite2D`` node has crossed the bottom line, it will emit the signal.
```